### ISSUE TITLE
PPO feed-forward for 'done' observation

### ISSUE THREAD

**Issue by cpwan** [opendilab/DI-engine#497]
- [x] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [x] RL algorithm bug
    + [ ] system worker bug
    + [ ] system utils bug
    + [ ] code design/refactor
    + [ ] documentation request
    + [ ] new feature request
- [x] I have visited the [readme](https://github.com/opendilab/DI-engine/blob/github-dev/README.md) and [doc](https://opendilab.github.io/DI-engine/)
- [x] I have searched through the [issue tracker](https://github.com/opendilab/DI-engine/issues) and [pr tracker](https://github.com/opendilab/DI-engine/pulls)
- [x] I have mentioned version numbers, operating system and environment, where applicable:
```bash
v0.4.2 1.10.0+cu102 3.8.0 | packaged by conda-forge | (default, Nov 22 2019, 19:11:38) 
[GCC 7.3.0] linux
```
---
Hi there, 
I found that the PPO did the feed-forward pass for both `obs` and `next_obs` when recomputing the advantage.
```python
# https://github.com/opendilab/DI-engine/blob/64f5dda5398063e81a632a1e3bbb1f3d1bc5bc7d/ding/policy/ppo.py#L170
value = self._learn_model.forward(data['obs'], mode='compute_critic')['value']
next_value = self._learn_model.forward(data['next_obs'], mode='compute_critic')['value']
```
When I print out the data, some of the `data['next_obs']` is the observation after the episode is 'done'.  However, when computing the GAE in a few lines afterward, it seems that the `next_value` at these 'done' observations is zeroed out [here](https://github.com/opendilab/DI-engine/blob/64f5dda5398063e81a632a1e3bbb1f3d1bc5bc7d/ding/rl_utils/gae.py#L55).

So, I guess the feed-forward pass on the 'done' observation is not necessary in the implementation. 

My problem is that:
**What if the feed-forward pass on the 'done' observation gives abnormal values, like inf/ nan?**

I am implementing a custom neural network model with action masks. When the environment is 'done', there is no valid action to perform and my neural network model would output 'nan'. As a consequence, the 'nan' entries in `next_value` cannot be zeroed out in the GAE and the training is ruined.

I feel that the feed-forward pass on the 'done' observation is ill-posed. How should I define it?

**Comment by PaParaZz1** [comment]
It is really a ill-posed problem especially in some environments you mentioned. Usually, many environments returns a normal but meaningless next state when this timestep id done, such as copy current state or zero state, so the computation of policy and value network will also be normal.

For you case, maybe you can modify your network to return `torch.zeros_like(value)` for the specific observation when done flag is true.
